## :warning: Failed tests

{{#if testFailures.length}}
  {{#each testFailures}}

<details>
<summary>
{{ testLocation }}
</summary>

<p/>

<!-- testLocation -->
[{{testLocation}}]({{ source_url testLocation }}) failed with error:

<!-- failureMessage -->
```
{{{failureMessage}}}
```

<!-- TODO: Test failureLocation -->
{{#if failureLocation}}
The error occurred at [{{failureLocation}}]({{ source_url failureLocation }}):
{{/if}}

{{#if testSnippet}}
{{#with testSnippet}}
<!-- TODO: Test testSnippet -->
```{{language}}
{{{codeFragment}}}
```
{{/with}}
{{/if}}

<!-- TODO: Test sourceDiff -->
{{#if sourceDiff }}
The following code changes may be related to this test failure:

```diff
{{{sourceDiff}}}
```
{{else}}
No relevant code changes found.
{{/if}}


| Diagram | Link |
| --- | --- |
<!-- TODO: sequence diagram diff, if available -->
{{!-- {{#if changedAppMap}}
{{#with changedAppMap}}
| Sequence diagram diff | {{> sequence-diagram-diff-link }} |
{{/with}}
{{/if}} --}}
| AppMap | [{{ appmap.name }}]({{ appmap_url 'head' appmap }}) |

</details>
  {{/each}}
{{/if}}
{{#else}}
All tests passed!
{{/else}}
